### DISCUSSION TITLE
When will the 2024 season go live in the addon?

### DISCUSSION THREAD

**Discussion by MauriceW67** [marcokreeft87/formulaone-card#331] [Q&A]
Title says it all :smile:

Will it happen automatically or does the addon need to be updated first?

**Comment by marcokreeft87** [comment]
It won't, the api stops unfortunately

**Comment by MauriceW67** [comment]
I don't understand. The Ergast API website says that it will work until the end of 2024? And the same is said in the issue topic:

https://github.com/marcokreeft87/formulaone-card/issues/299#issuecomment-1750055432

Also, if I browse to http://ergast.com/api/f1/2024 I can see the correct information for this season?

**Reply by marcokreeft87** [reply]
Oh weird. I used to say till the end of 2023

**Reply by MauriceW67** [reply]
Not in the issue that I quoted 😄 

You even wrote yourself "We got one year left"

But regardless, the addon does not display 2024 automatically. Or is there a hard limit programmed in?

**Reply by marcokreeft87** [reply]
> Not in the issue that I quoted 😄 
> 
> You even wrote yourself "We got one year left"
> 
> But regardless, the addon does not display 2024 automatically. Or is there a hard limit programmed in?
> 

Oh damn haha then I don't know where I got it from xD But the schedule is cached for a month, so somewhere around the 7th it should be working again

**Comment by MauriceW67** [comment]
> Oh damn haha then I don't know where I got it from xD But the schedule is cached for a month, so somewhere around the 7th it should be working again

Ok thanks, then we will see what happens next week 😃

**Comment by marcokreeft87** [comment]
The problem is the API. It's not providing the start time of the races. I will try and fix it on friday

**Comment by marcokreeft87** [comment]
Sorry guys, but the data of the current season is useless from the API. 
![image](https://github.com/marcokreeft87/formulaone-card/assets/10223677/91704959-b6c8-444a-81d3-baaa12582e96)
This of from last year
![image](https://github.com/marcokreeft87/formulaone-card/assets/10223677/b30d2e8a-f9bd-4108-9fdf-d260af9f976f)
This is from this year.

As you can see the practice times are gone and the qualifying.. I have to rewrite all the cards to be able to use this data and even than it won't be more than a glorified table without the practice sessions and stuff.

I will continue to check for updates. Lets hope this changes!

**Reply by JeroenvanA** [reply]
Sorry to hear that Marco. I hope you find a solution without to much work

**Reply by acla6525** [reply]
Could this be because the API “current” link is still on 2023? http://ergast.com/api/f1/current

Once it updates to 2024 there may be more info. 

There are also a few people on the API discussion forum seemingly close to developing a new API that will run from 2024 onwards.

**Reply by marcokreeft87** [reply]
Nice! Will you post it here when those apis are available?

**Reply by MauriceW67** [reply]
@acla6525 Where can we find the "API discussion forum"? Do you have a link?

**Reply by acla6525** [reply]
It’s under feedback, a few people talking about taking over the API and doing their own. 
http://ergast.com/mrd/feedback/

**Comment by JeroenvanA** [comment]
Is this usable?  https://openf1.org/

**Reply by marcokreeft87** [reply]
Nope, thats only historical data

**Comment by MauriceW67** [comment]
@marcokreeft87 There is also a slack channel where some people from FastF1 are discussing the creation of a new API:

https://fastf1.slack.com/join/shared_invite/zt-25osjniqp-TsBgxY_7alSMar5wEgWcXQ#/shared-invite/email

Not sure how far along their project is though. The slack channel is not really busy as far as I can tell.

**Comment by Sladgela** [comment]
Is this what we were looking for?

https://ergast.com/api/f1/2024

**Comment by kirsty99999** [comment]
> Title says it all 😄
> 
> Will it happen automatically or does the addon need to be updated first?

Hi, just checking in on the status of this? I love my F1 dashboard as does my son. Hoping it will become available this season too. Thanks!

**Comment by snhnic** [comment]
Hopefully it will reset for season 2024

**Comment by MauriceW67** [comment]
It seems that now that the first race is over, the API is working a bit better. I can also see that for the first 10 races the session start times have been added. The only thing that doesn't load properly yet is the schedule card. But that's probably because not all session start times have been added yet?

**Reply by kirsty99999** [reply]
I see 'error getting the schedule'. I can see the next race in the phone app (iphone), but shows as 'Season Over, see you next year' on my PC Browser.

**Comment by MauriceW67** [comment]
@kirsty99999 For me it works in my PC browser. Did you try to do a Ctrl-F5 (hard refresh) on the page?

I can see: Next race, Previous race result, Driver standings and Constructor standings. Only Race schedule does not work.

**Reply by marcokreeft87** [reply]
They all have separate cache. Can take up to 3 days (schedule for example)

**Comment by MauriceW67** [comment]
> They all have separate cache. Can take up to 3 days (schedule for example)

If I check the API page (https://ergast.com/api/f1/current) I now see session times for all races. So hopefully the schedule will start to work in a couple of days from now.

**Comment by everdrift** [comment]
Since the ergast API is quite unreliable these days maybe this is an option: https://github.com/f1db/f1db ?
He doesn't offer an API yet (its on his TODO list) but json and SQLite database is available.

**Comment by acla6525** [comment]
I have no issues with all the data for current season now that the first race is over. All cards are working for me.

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Reply by everdrift** [reply]
hmmmmm...... right now only I can only see ''standings" all the others show "season is over, see you next year"
This is on all my devices (Windows, MacOS and iOS)

**Comment by acla6525** [comment]
I only installed the add on a few months ago, but everything shows up across all dashboards/devices.

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Reply by everdrift** [reply]
only the card so not the Formula One API integration?

**Reply by marcokreeft87** [reply]
It's a card guys,  no addon, no integration.
But there is a Refresh cache button (check the wiki)

**Comment by marcokreeft87** [comment]
Since everything is back to normal on the api side, I'm closing this thread
